core/commands/get.go

```diff
@@ -238,13 +240,50 @@ func (gw *getWriter) writeArchive(r io.Reader, fpath string) error {
 
 func (gw *getWriter) writeExtracted(r io.Reader, fpath string) error {
 	fmt.Fprintf(gw.Out, "Saving file(s) to %s\n", fpath)
+	var modified bool
+	defer func() { //NOTE: defer order important; we want this to display after the progressbar finishes
+		if modified {
+			fmt.Fprintf(gw.Out, "WARNING: Extraction output had to be modified in order to be stored successfully\n")
+		}
+	}()
+
 	bar := makeProgressBar(gw.Err, gw.Size)
 	bar.Start()
 	defer bar.Finish()
 	defer bar.Set64(gw.Size)
 
 	extractor := &tar.Extractor{Path: fpath, Progress: bar.Add64}
-	return extractor.Extract(r)
+	if osh.IsWindows() {
+		extractor.Sanitize(true)
+		builtinSanitizer := extractor.SanitizePathFunc
+		extractor.SanitizePathFunc = func(path string) (string, error) {
```

Comment:
also, if not too difficult, making these anonymous functions into real functions would improve readability here a good deal

Reply:
actually, just doing this should clean everything up well enough